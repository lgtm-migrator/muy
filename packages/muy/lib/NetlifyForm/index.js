import React from "react"
import PropTypes from "prop-types"
import Form from "../Form"

const NetlifyForm = ({ action, children, honeypot, id }) => {
  const honeyPotName = id ? `${id}-bot-field` : `bot-field`
  return (
    <Form
      action={action}
      data-netlify={"true"}
      data-netlify-honeypot={honeypot && honeyPotName}
      id={id}
      name={id}
    >
      {honeyPotName && (
        <input
          aria-hidden={true}
          id={honeyPotName}
          name={honeyPotName}
          type={"hidden"}
        />
      )}
      <input
        aria-hidden={true}
        id={"form-name"}
        name={"form-name"}
        type={"hidden"}
      />
      {children}
    </Form>
  )
}

export default NetlifyForm

NetlifyForm.defaultProps = {
  honeypot: true,
  id: "contact",
  name: "Contact form",
}

NetlifyForm.propTypes = {
  action: PropTypes.string,
  honeypot: PropTypes.bool,
  id: PropTypes.string,
  name: PropTypes.string,
}
