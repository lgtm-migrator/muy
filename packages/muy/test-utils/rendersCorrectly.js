import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
