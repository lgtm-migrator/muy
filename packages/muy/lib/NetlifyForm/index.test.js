import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import NetlifyForm from "."

describe("<NetlifyForm/> component", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<NetlifyForm />)
  })
})
