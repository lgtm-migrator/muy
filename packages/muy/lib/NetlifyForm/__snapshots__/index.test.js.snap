// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetlifyForm/> component should render correctly with no props 1`] = `
<form
  className="MuiBox-root MuiBox-root-2"
  data-name="contact"
  data-netlify="true"
  data-netlify-honeypot="contact-bot-field"
  id="contact"
  name="contact"
>
  <input
    aria-hidden={true}
    id="contact-bot-field"
    name="contact-bot-field"
    type="hidden"
  />
  <input
    aria-hidden={true}
    id="form-name"
    name="form-name"
    type="hidden"
  />
</form>
`;
